sopel/bot.py

```diff
@@ -305,18 +305,28 @@ def say(self, text, recipient, max_messages=1):
 
             recipient_id = Identifier(recipient)
 
-            if recipient_id not in self.stack:
-                self.stack[recipient_id] = []
-            elif self.stack[recipient_id]:
-                elapsed = time.time() - self.stack[recipient_id][-1][0]
-                if elapsed < 3:
-                    penalty = float(max(0, len(text) - 40)) / 70
-                    wait = 0.8 + penalty
-                    if elapsed < wait:
-                        time.sleep(wait - elapsed)
+            reciprec = self.stack.get(recipient_id)
```

Comment:
I think @Exirel would agree with me that we should avoid shortcutting references to `self.stack`. It makes the code more confusing to come back and reread later.

Hell, I'm already confused trying to remember what `reciprec` stands for while reviewing. At the very least, it should have a less cryptic name.

Reply:
I don't mind the shortcut, as long as the name makes sense. In this case, I agree with @dgw that `reciprec` can be confusing. It's more a `last_messages`, a `recipient_info`, or just a `recipient`. While we talk about "naming things', I think `burst` could be `flood_left`, or `flood_count`, or `flood_token`, or something alike. I don't know, I'm not that good.

Last, you can replace the get + if with this one operation:

```
recipient = self.stack.setdefault(recipient_id, {
    'messages': [],
    'flood_left': self.config.core.flood_burst_lines,
})
```